.github/workflows/actions.yml

```diff
@@ -0,0 +1,114 @@
+name: innovativeproject-loancalculator
+
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: '**'
+
+jobs:
+  lint-backend:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.9.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.9.10"
+    
+    - name: Install backend dependencies
+      run: |
+        curl -sSL https://install.python-poetry.org | python3 -
+        poetry config virtualenvs.create false
+        cd backend
+        poetry install
+    
+    - name: Format and lint backend with Black
+      run: |
+        black .    
+    
+  test-backend:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.9.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.9.10"
+    
+    - name: Install backend dependencies
+      run: |
+        curl -sSL https://install.python-poetry.org | python3 -
+        poetry config virtualenvs.create false
+        cd backend
+        poetry install
+    
+    - name: Test backend with pytest
+      run: |
+        pytest
+  
+  lint-frontend:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Set up Node 16.14.0
```

Comment:
Setup for node deps should be done once and it should use caching mechanism like this one: https://github.com/actions/setup-node